spec/core/ics-033-multi-hop/README.md

```diff
@@ -0,0 +1,571 @@
+//         proof          - The serialized multihop proof data.
+//         prefix         - Merkleprefix to be combined with key to generate Merklepath for the key/value proof verification.
+//         key            - The key to prove in the indexed consensus state.
+//         value          - The value to prove in the indexed consensus state.
+func VerifyMultihopMembership(
+    consensusState exported.ConsensusState,
+    connectionHops []string,
+    proof MultihopProof,
+    prefix exported.Prefix,
+    key string,
+    value []byte,
+) {
+    // deserialize proof bytes into multihop proofs
+    proofs := abortTransactionUnless(Unmarshal(proof))
+    abortTransactionUnless(len(proofs.ConsensusProofs) >= 1)
+    abortTransactionUnless(len(proofs.ConnectionProofs) == len(proofs.ConsensusProofs))
+
+    // verify connection hop ordering and connections are in OPEN state
+    abortTransactionUnless(VerifyConnectionHops(proofs.ConnectionProofs, connectionHops))
+
+    // verify intermediate consensus and connection states from receiver --> sender
+    abortTransactionUnless(VerifyConsensusAndConnectionStates(consensusState, proofs.ConsensusProofs, proofs.ConnectionProofs))
+
+    // verify a key/value proof on source chain's consensus state.
+    abortTransactionUnless(VerifyKeyMembership(consensusState, proofs, prefix, key, value))
+}
+
+// VerifyMultihopNonMembership verifies a multihop non-membership proof.
+// Inputs: consensusState - The consensusState for chain[1], which is known on the receiving chain (chain[0]).
+//         connectionHops - The expected connectionHops for the channel from the receiving chain to the sending chain.
+//         proof          - The serialized multihop proof data.
+//         prefix         - Merkleprefix to be combined with key to generate Merklepath for the key/value proof verification.
+//         key            - The key to prove absent in the indexed consensus state
+func VerifyMultihopNonMembership(
+    consensusState exported.ConsensusState,
+    connectionHops []string,
+    proof MultihopProof,
+    prefix exported.Prefix,
+    key string,
+) {
+    // deserialize proof bytes into multihop proofs
+    proofs := abortTransactionUnless(Unmarshal(proof))
+    abortTransactionUnless(len(proofs.ConsensusProofs) >= 1)
+    abortTransactionUnless(len(proofs.ConnectionProofs) == len(proofs.ConsensusProofs))
+
+    // verify connection hop ordering and connections are in OPEN state
+    abortTransactionUnless(VerifyConnectionHops(proofs.ConnectionProofs, connectionHops))
+
+    // verify intermediate consensus and connection states from receiver --> sender
+    abortTransactionUnless(VerifyIntermediateStateProofs(consensusState, proofs.ConsensusProofs, proofs.ConnectionProofs))
+
+    // verify a key/value proof on source chain's consensus state.
+    abortTransactionUnless(VerifyKeyNonMembership(consensusState, proofs, prefix, key))
```

Comment:
related to the previous comment we could have `VerifyMultiHopProofs` or something like that, would verify both the connection proof and consensus proof in in one step of the iterator.
Can even change `MultihopProof` to be a slice of (connection_proof, cosnensus_proof) elements. 
Do you have a pointer to the .protos for these in your code?